src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1836,19 +1836,39 @@ def derive_scrypt(self, key_material, salt, length, n, r, p):
         return self._ffi.buffer(buf)[:]
 
     def chacha20poly1305_encrypt(self, key, nonce, data, associated_data):
-        return chacha20poly1305.encrypt(
-            self, key, nonce, data, associated_data
+        return aead._encrypt(
+            self, b"chacha20-poly1305", key, nonce, data, associated_data, 16
         )
 
     def chacha20poly1305_decrypt(self, key, nonce, data, associated_data):
-        return chacha20poly1305.decrypt(
-            self, key, nonce, data, associated_data
+        return aead._decrypt(
+            self, b"chacha20-poly1305", key, nonce, data, associated_data, 16
         )
 
-    def chacha20poly1305_supported(self):
+    def aesccm_encrypt(self, key, nonce, data, associated_data, tag_length):
+        cipher_name = "aes-{0}-ccm".format(len(key) * 8).encode("ascii")
+        return aead._encrypt(
+            self, cipher_name, key, nonce, data, associated_data, tag_length
+        )
+
+    def aesccm_decrypt(self, key, nonce, data, associated_data, tag_length):
+        cipher_name = "aes-{0}-ccm".format(len(key) * 8).encode("ascii")
+        return aead._decrypt(
+            self, cipher_name, key, nonce, data, associated_data, tag_length
+        )
+
+    def aead_cipher_supported(self, cls):
+        # We do this ugly thing because circular import problems prevent
+        # importing aead so we could just do isinstance checks
+        if cls.__name__ == "ChaCha20Poly1305":
```

Comment:
Circular import because we import backend in cryptography.hazmat.primitives.ciphers.aead :(